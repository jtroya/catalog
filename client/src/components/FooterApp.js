import React from 'react';
import { Footer } from 'react-materialize';

function FooterApp() {
  return (
      <Footer className="footer_app" copyrights="2018 Copyright Text" />
  );
}

export default FooterApp;
