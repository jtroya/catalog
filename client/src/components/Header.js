import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Header({ appName }) {
  return (
    <nav>
      <div className="nav-wrapper">
        <Link to="/"><span className="brand-logo center">{ appName }</span></Link>
      </div>
    </nav>
  );
};

Header.propTypes = {
  appName: PropTypes.string  
};

Header.defaultProps = {
  appName: 'Catalog'  
};

export default Header;
