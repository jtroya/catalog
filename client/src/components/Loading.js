import React from 'react';
import PropTypes from 'prop-types';
import { Preloader, Row, Col } from 'react-materialize';

function Loading({ flashing, size }) {
  return (
      <Row className="container">
        <Col s={ 12 } m={ 12 }>
          <div className="center-align">
            <Preloader flashing={ flashing } size={ size } />
          </div>
        </Col>
      </Row>
  );
}

Loading.propTypes = {
  flashing: PropTypes.bool,
  size: PropTypes.string
};

Loading.defaultProps = {
  flashing: false,
  size: 'medium'
};

export default Loading;
