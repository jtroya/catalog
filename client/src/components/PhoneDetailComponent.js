import React, { Component } from 'react';
import { Row, Col, Table } from 'react-materialize';

class PhoneDetailComponent extends Component {
  componentDidMount () {
    this.props.getInfoDetail(this.props.phoneId);
  }

  renderInfo = () => {
    const { brand, model, color, description, price, storage, imageUrl  } = this.props.selectedItem[0];
    return (
      <Row className="container">
        <Col s={ 12 } m={ 4 } l={ 4 }>
          <div className="div_panel scale-transition">
            <img className="div_panel-image" alt={ model } src={ imageUrl } />
          </div>
        </Col>
        <Col s={ 12 } m={ 8 } l={ 8 }>
          <Table>
            <tbody>
              <tr><td>Model</td><td>{ model }</td></tr>
              <tr><td>Brand</td><td>{ brand }</td></tr>
              <tr><td>Colour</td><td>{ color }</td></tr>
              <tr><td>Description</td><td>{ description }</td></tr>
              <tr><td>Price</td><td>{ price }</td></tr>
              <tr><td>Storage</td><td>{ storage }</td></tr>
            </tbody>
          </Table>
        </Col>
      </Row>
    );
  }

  render() {    
    return (
      <Row>
        { this.props.selectedItem[0] &&
          this.renderInfo()
        }
      </Row>
    )
  }
}

export default PhoneDetailComponent;
