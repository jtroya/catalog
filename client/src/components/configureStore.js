import { createStore, applyMiddleware } from 'redux'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage' // defaults to localStorage for web and AsyncStorage for react-native
import reduxThunk from 'redux-thunk'

import reducers from '../reducers'

const persistConfig = {
  key: 'root',
  storage,
}

const persistedReducer = persistReducer(persistConfig, reducers)

export default () => {
  let store = createStore(
    persistedReducer,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
    applyMiddleware(reduxThunk)
  )
  let persistor = persistStore(store)
  return { store, persistor }
}
