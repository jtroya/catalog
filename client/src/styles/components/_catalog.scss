.footer_app {
  right: 0;
  bottom: 0;
  left: 0;
  padding: 1rem;
}

.card_img {
  height: 180px;
  margin: 10px 10px;

  img {
    height: 120px;
  }
}

.div_panel {
  height: 300px;
}

.div_panel-image {
  height: 220px;
  width: auto;
}
