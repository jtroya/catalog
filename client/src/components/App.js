import React, { Component } from 'react';
import { Row } from 'react-materialize';
import { BrowserRouter, Route } from 'react-router-dom';

import Header from './Header';
import FooterApp from './FooterApp';
import PhoneListContainer from '../containers/PhoneListContainer';
import PhoneDetailContainer from '../containers/PhoneDetailContainer';

export class App extends Component {
  render() {
      return (
        <BrowserRouter>
          <Row>
            <Header />
            <div className="section">
              <Route exact path="/" component= { PhoneListContainer } />
              <Route exact path="/detail/:phoneId" render= { ({ match }) => (
                <PhoneDetailContainer phoneId={ match.params.phoneId } />
              )} />
            </div>
            <FooterApp />
          </Row>
        </BrowserRouter>
      );
  }
}

export default App;
