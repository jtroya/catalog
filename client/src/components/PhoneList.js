import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Row, Col, Card, CardTitle } from 'react-materialize';
import { Link } from 'react-router-dom';

import Loading from './Loading';

class PhoneList extends Component {
  componentDidMount() {
    this.props.fetchCatalog();
  }
  
  handleLinkDetail = (goTo, value) => {
    return (
      <Link to={ goTo }>{ value }</Link>
    );
  };

  displayHeader = (value) => {
    return (
      <CardTitle className="card_img" image={value}></CardTitle>
    );
  };

  renderList () {
    const { catalog } = this.props;
    if (catalog) {
      return catalog.map((item, index) => {
        return (
          <Col s={ 12 } m={ 6 } l={ 4 } key={ index }>
            <Card horizontal className="hoverable"
              header={ this.displayHeader(item.imageUrl)}
              title={ item.model}>
              { this.handleLinkDetail(`/detail/${item.phoneId}`, item.description) }
            </Card>
          </Col>
        );
      });
    }
  }
  
  render() {
    const { loading } = this.props;
    return (
      <div className="container">
        <Row>
          { loading &&
            <Loading size='big' /> }
          { !loading &&
            this.renderList()
          }
        </Row>
      </div>
    )
  }
}

PhoneList.propTypes = {
  loading: PropTypes.bool,
  catalog: PropTypes.array
};

export default PhoneList;
