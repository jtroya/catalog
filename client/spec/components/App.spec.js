import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { App } from '../../src/components/App';

describe('App', () => {
    let wrapper;

    it('should render de App', () => {
        wrapper = shallow(<App />);
        expect(wrapper.find('Row')).toBeTruthy();
        expect(wrapper.find('Header')).toBeTruthy();
        expect(wrapper.find('FooterApp')).toBeTruthy();
    });

    it('matches the snapshot', () => {
        const tree = shallow(<App />);
        expect(toJson(tree)).toMatchSnapshot();
      })
});
