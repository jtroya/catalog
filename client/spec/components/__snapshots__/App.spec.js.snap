// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<BrowserRouter>
  <Row>
    <Header
      appName="Catalog"
    />
    <div
      className="section"
    >
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        exact={true}
        path="/detail/:phoneId"
        render={[Function]}
      />
    </div>
    <FooterApp />
  </Row>
</BrowserRouter>
`;
